lib/awful/tooltip.lua

```diff
@@ -405,6 +412,21 @@ function tooltip:set_preferred_positions(value)
     set_geometry(self)
 end
 
+--- Set a widget to display.
+-- @tparam widget widget The widget to display
+-- @tparam integer width The width of the tooltip
+-- @tparam integer height The height of the tooltip
+function tooltip:set_widget(widget, width, height)
```

Comment:
I think it should be 3 properties, but a function with 3 parameters. The size concerns and widgets concerns seems rather different. One might want different size with the text and others might be ok with the size returned by `:fit()` as-is (with the screen width/height as inputs)